src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/AddExtendedQueryTagService.cs

```diff
@@ -30,7 +34,7 @@ public class AddExtendedQueryTagService : IAddExtendedQueryTagService
             EnsureArg.IsNotNull(extendedQueryTagEntryValidator, nameof(extendedQueryTagEntryValidator));
             EnsureArg.IsNotNull(extendedQueryTagConfiguration?.Value, nameof(extendedQueryTagConfiguration));
 
-            _extendedQueryTagStore = extendedQueryTagStoreFactory.GetInstance();
+            _extendedQueryTagStore = new Lazy<IExtendedQueryTagStore>(() => extendedQueryTagStoreFactory.GetInstance());
```

Comment:
With this service probably being "scoped," it seems strange to also have a nested `Lazy<T>`. Couldn't we resolve the instance via DI every time this service is resolved?

Reply:
I didn't find a different way out so far.
In the context of Dicom Server, these components (`AddExtendedQueryTagService` etc.) are initialized after the host service which initialize database and update schemaInformation correctly. In Azure Function , we don't have such things.

Reply:
Ultimately, can we instead call the factory in the methods to resolve the proper store instead of in the ctor? Sort of like the `HttpClientFactory`?

Reply:
I haven't figure out a good way for this.
The underlying reason is that DicomServer and DicomFunctions(the durable function) initialize ExtendedQueryTagStore in different ways -- DicomServer has a hostedService which initialize SQL before `AddExtendedQueryTagService` is initialized, so `SchemaInformation.CurrentVersion` already have the version. For DicomFunctions, we have to set up the version as part of orchestration, so that value always be `Null` when DI.
That means `SqlStoreFactory` doesn't fit for both case, we may need a different implementation for DicomFunctions.

Reply:
Will's PR https://github.com/microsoft/dicom-server/pull/843 should solve this problem.